sdk/include/xdk/ketypes.h

```diff
@@ -1316,31 +1353,71 @@ typedef struct _KUSER_SHARED_DATA
         } DUMMYSTRUCTNAME;
     } DUMMYUNIONNAME;
     UCHAR TscQpcPad[2];                                     // 0x2ee
+#elif (NTDDI_VERSION >= NTDDI_WIN10_RS1)
+    union
+    {
+        UCHAR VirtualizationFlags;                          // 0x2ed
+#if defined(_ARM64_)
+        struct
+        {
+            UCHAR ArchStartedInEl2 : 1;
+            UCHAR QcSlIsSupported : 1;
+            UCHAR : 6;
+        };
 #endif
+    };
+    UCHAR Reserved12[2];                                    // 0x2ee
+#else
+    UCHAR Reserved12[3];                                    // 0x2ed
+#endif // NTDDI_VERSION == NTDDI_WIN7
+
 #if (NTDDI_VERSION >= NTDDI_VISTA)
     union
     {
         ULONG SharedDataFlags;                              // 0x2f0
         struct
         {
-            ULONG DbgErrorPortPresent:1;                    // 0x2f0
-            ULONG DbgElevationEnabled:1;                    // 0x2f0
-            ULONG DbgVirtEnabled:1;                         // 0x2f0
-            ULONG DbgInstallerDetectEnabled:1;              // 0x2f0
-            ULONG DbgSystemDllRelocated:1;                  // 0x2f0
-            ULONG DbgDynProcessorEnabled:1;                 // 0x2f0
-            ULONG DbgSEHValidationEnabled:1;                // 0x2f0
-            ULONG SpareBits:25;                             // 0x2f0
+            ULONG DbgErrorPortPresent : 1;                  // 0x2f0
+            ULONG DbgElevationEnabled : 1;                  // 0x2f0
+            ULONG DbgVirtEnabled : 1;                       // 0x2f0
+            ULONG DbgInstallerDetectEnabled : 1;            // 0x2f0
+#if (NTDDI_VERSION >= NTDDI_WIN8)
+            ULONG DbgLkgEnabled : 1;                        // 0x2f0
+#else
+            ULONG DbgSystemDllRelocated : 1;                // 0x2f0
+#endif
+            ULONG DbgDynProcessorEnabled : 1;               // 0x2f0
+#if (NTDDI_VERSION >= NTDDI_WIN8)
+            ULONG DbgConsoleBrokerEnabled : 1;              // 0x2f0
+#else
+            ULONG DbgSEHValidationEnabled : 1;              // 0x2f0
+#endif
+            ULONG DbgSecureBootEnabled : 1;                 // 0x2f0 Win8+
+            ULONG DbgMultiSessionSku : 1;                   // 0x2f0 Win 10+
+            ULONG DbgMultiUsersInSessionSku : 1;            // 0x2f0 Win 10 RS1+
+            ULONG DbgStateSeparationEnabled : 1;            // 0x2f0 Win 10 RS3+
+            ULONG SpareBits                 : 21;           // 0x2f0
```

Comment:
Not sure you need to repeat "0x2f0" everytime for all these! (see the other union struct bitfields above)